### PR TITLE
zos,fs: check F_OK separately in uv_fs_access()

### PR DESCRIPTION
**Pull request by zsw007**
On z/OS, `F_OK` is 8 instead of 0. This PR creates an internal `uv__fs_access()` function, where the access check for `F_OK` needs to be separated from `R_OK`, `W_OK`, and `X_OK` on z/OS. Otherwise, checking that a file grants read and write permissions using `F_OK | R_OK | W_OK` will fail.

Also updates the `fs_access` test to include this edge case.

### CODE REVIEW

File: src/unix/fs.c [hunk 1]
```diff
@@ -1680,6 +1680,22 @@ static ssize_t uv__fs_write_all(uv_fs_t* req) {
 }
 
 
+#ifdef __MVS__
+static ssize_t uv__fs_access(uv_fs_t* req) {
+  /* On z/OS, F_OK is 8 instead of 0, so separate the access checks here. */
```
Comment (cjihrig) [inline src/unix/fs.c:1685]: It's not immediately clear (at least to me) just from reading this comment why `F_OK` being 8 requires special treatment. The docs I found say:

> If you are using F_OK to test for the file's existence, you cannot use OR with any of the other symbols.

Maybe the comment could just say "`F_OK` cannot be OR'ed with other access modes on z/OS."

Comment (richardlau) [reply src/unix/fs.c:1685]: `F_OK` not being allowed to be OR'ed with other access modes does not appear to be a z/OS specific limitation.

e.g. https://linux.die.net/man/2/access
> The mode specifies the accessibility check(s) to be performed, and is either the value F_OK, or a mask consisting of the bitwise OR of one or more of R_OK, W_OK, and X_OK.

and https://pubs.opengroup.org/onlinepubs/9699919799/functions/access.html
> The value of amode is either the bitwise-inclusive OR of the access permissions to be checked (R_OK, W_OK, X_OK) or the existence test (F_OK).

Comment (cjihrig) [reply src/unix/fs.c:1685]: I should have checked the docs for other platforms 🤦 ... in that case, do we even need to do anything here?

File: test/test-fs.c [hunk 2]
```diff
@@ -1580,6 +1580,19 @@ TEST_IMPL(fs_access) {
   ASSERT(access_cb_count == 1);
   access_cb_count = 0; /* reset for the next test */
 
+  /* File should grant read and write permission */
+  r = uv_fs_access(NULL, &req, "test_file", F_OK | R_OK | W_OK, NULL);
```
Comment (richardlau) [inline test/test-fs.c:1584]: My reading of https://linux.die.net/man/2/access and https://pubs.opengroup.org/onlinepubs/9699919799/functions/access.html suggests that the `F_OK | R_OK | W_OK` combination of flags is not valid (not just on z/OS) and not expected to work. @zsw007 Is there some context as to the reason this PR was created -- perhaps z/OS was doing the right thing all along?

Comment (zsw007) [reply test/test-fs.c:1584]: In Node.js, under the parallel/test-fs-access test case, `F_OK | R_OK | W_OK` is being tested for:

https://github.com/nodejs/node/blob/162cc4fa6303e3dd3a1f7ec041f9aa073a40d7d7/test/parallel/test-fs-access.js#L156

This Node.js test case currently passes on platforms other than z/OS, and the changes in this PR is needed for the test case to pass on z/OS as well. The extra test case that I added to this PR is based off that Node.js test case (which currently also passes on platforms other than z/OS).

But if you're saying that `F_OK | R_OK | W_OK` is not valid, then I'm wondering what's the reason for it being tested in Node.js?

Comment (richardlau) [reply test/test-fs.c:1584]: > In Node.js, under the parallel/test-fs-access tests, `F_OK | R_OK | W_OK` is being tested for on line 156:
> 
> https://github.com/nodejs/node/blob/162cc4fa6303e3dd3a1f7ec041f9aa073a40d7d7/test/parallel/test-fs-access.js#L156
> 
> This Node.js test case currently passes on platforms other than z/OS, and the changes in this PR is needed for the test case to pass on z/OS as well. The extra test case that I added to this PR is based off that Node.js test case (which currently also passes on platforms other than z/OS, and needs this PR to pass on z/OS).
> 
> But if you're saying that `F_OK | R_OK | W_OK` is not valid, then I'm wondering what's the reason for it being tested in Node.js?

The Node.js test appears to have been added in https://github.com/nodejs/node/pull/114 which was authored by... @cjihrig 😄.

Comment (cjihrig) [reply test/test-fs.c:1584]: Past Colin comes back to haunt me again. I'll open a PR to fix the Node.js test.

Comment (zsw007) [reply test/test-fs.c:1584]: Ah ok, so if ORing F_OK is a mistake in that test case, then it looks like this PR isn't necessary and the test case can be fixed instead 🙂
